### PR TITLE
Community: VectorStore Infinispan. Adding TLS and authentication

### PR DESCRIPTION
**Pull request by rigazilla**
**Description**:
this PR enable VectorStore TLS and authentication (digest, basic) with HTTP/2 for Infinispan server.
Based on httpx.

Added docker-compose facilities for testing
Added documentation

**Dependencies:**
requires `pip install httpx[http2]` if HTTP2 is needed

**Twitter handle:**
https://twitter.com/infinispan

### CODE REVIEW

**Comment by rigazilla** [comment]
@baskaryan could you have a look pls?

**Comment by rigazilla** [comment]
@baskaryan  or @efriis please checks are all OK

**Comment by rigazilla** [comment]
anyone can comment this PR?
@baskaryan, @efriis, @eyurtsev, @ccurme, @vbarda, @hwchase17 please

File: libs/community/langchain_community/vectorstores/infinispanvs.py [hunk 1]
```diff
@@ -58,13 +57,34 @@ def __init__(
...
+        kwargs: Any
+            Rest of arguments passed to Infinispan. See docs"""
         self.ispn = Infinispan(**kwargs)
         self._configuration = kwargs
         self._cache_name = str(self._configuration.get("cache_name", "vector"))
         self._entity_name = str(self._configuration.get("entity_name", "vector"))
         self._embedding = embedding
-        self._textfield = self._configuration.get("textfield", "text")
-        self._vectorfield = self._configuration.get("vectorfield", "vector")
+        self._textfield = self._configuration.get("text_field", "text")
```
Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/infinispanvs.py:86]: is this a breaking change or a bugfix?

Comment (baskaryan) [inline libs/community/langchain_community/vectorstores/infinispanvs.py:86]: whats the motivation here? this is a breaking change

Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py:86]: I wanted to uniform the params naming using _ everywhere.
I can change the code to accept both formats and warn than old format will be removed in the future

Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py]: @baskaryan done

Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py]: @baskaryan pls

File: libs/community/langchain_community/vectorstores/infinispanvs.py [hunk 2]
```diff
@@ -382,19 +438,68 @@ class Infinispan:
     """
 
     def __init__(self, **kwargs: Any):
```
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/infinispanvs.py:440]: nit: no change required here, but masking all the inputs in **kwargs** and unpacking in the init body instead of just doing it in the init function signature is fairly uncommon  as it introduces unintended bugs more often than not

File: libs/community/langchain_community/vectorstores/infinispanvs.py [hunk 3]
```diff
@@ -472,15 +573,15 @@ def put(self, key: str, data: str, cache_name: str) -> requests.Response:
...
             api_url,
-            data,
+            content=data,
             headers={"Content-Type": "application/json"},
             timeout=REST_TIMEOUT,
         )
         return response
 
-    def get(self, key: str, cache_name: str) -> requests.Response:
+    def get(self, key: str, cache_name: str) -> Any:
```
Comment (eyurtsev) [inline libs/community/langchain_community/vectorstores/infinispanvs.py:584]: why the change to `Any` from `requests.Response`?

File: libs/community/langchain_community/vectorstores/infinispanvs.py [hunk 4]
```diff
@@ -382,19 +438,68 @@ class Infinispan:
     """
 
     def __init__(self, **kwargs: Any):
+        """
+        Parameters
+        ----------
+        schema: str
+            Schema for HTTP request: "http" or "https". Default "http"
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py:458]: just to keep things simple for the first try, users will be happy to deal with tls certs if they need security

File: libs/community/langchain_community/vectorstores/infinispanvs.py (hunk 2 above)
Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py]: thanks, I unpacked the signature

File: libs/community/langchain_community/vectorstores/infinispanvs.py (hunk 3 above)
Comment (rigazilla) [reply libs/community/langchain_community/vectorstores/infinispanvs.py]: when I started this PR the httpx lib was optional, so the idea was to allow both httpx.Response and request.Response. But now there's only httpx.Response.
Changed, thanks!

**Comment by rigazilla** [comment]
@eyurtsev @baskaryan any chance to have this merged? Or at least a reason why it can't?

File: libs/community/tests/integration_tests/vectorstores/test_infinispanvs.py [hunk 5]
```diff
@@ -1,7 +1,9 @@
 """Test Infinispan functionality."""
```
Comment (eyurtsev) [inline libs/community/tests/integration_tests/vectorstores/test_infinispanvs.py:1]: FYI: There's a standard testing suite for vectorstores in langchain_standard_tests, it'll likely weed out a number of bugs

**Comment by eyurtsev** [comment]
@rigazilla try to DM, w/ existing volume + how PR notifications work it's hard to keep track, we're trying to transition to using the projects PR preview to avoid dropping the ball on PRs being reviewed.

Generally, we're going to be prioritizing community PRs based on community support for them (:+1: emojis).

**Comment by rigazilla** [comment]
thanks @eyurtsev
